server/mongodb/actions/Analytics.ts

```diff
@@ -123,3 +124,256 @@ export const updateSessionComplete = async (
 
   return data;
 };
+
+// average weekly stats function
+export const averageWeeklyStats = async () => {
+  // divide all values by completed sessions and added streaklength
+  await Analytics.updateMany({}, [
+    {
+      $set: {
+        weeklyMetrics: {
+          $map: {
+            input: {
+              $range: [0, { $size: "$weeklyMetrics" }],
+            },
+            in: {
+              $cond: [
+                { $eq: ["$$this", 0] },
+                {
+                  $mergeObjects: [
+                    { $arrayElemAt: ["$weeklyMetrics", "$$this"] },
+                    {
+                      streakLength: { $size: "$streak" },
+                      math: {
+                        $let: {
+                          vars: {
+                            firstItem: {
+                              $arrayElemAt: ["$weeklyMetrics", 0],
+                            },
+                          },
+                          in: {
+                            sessionsCompleted:
+                              "$$firstItem.math.sessionsCompleted",
+                            questionsAttempted: {
+                              $cond: [
+                                {
+                                  $eq: [
+                                    "$$firstItem.math.sessionsCompleted",
+                                    0,
+                                  ],
+                                },
+                                0,
+                                {
+                                  $divide: [
+                                    "$$firstItem.math.questionsAttempted",
+                                    "$$firstItem.math.sessionsCompleted",
+                                  ],
+                                },
+                              ],
```

Comment:
Since this condition is repeated many times in this query with different parameters, can you create a function called `getDivideQuery(numeratorField: string, denominatorField: string)` that builds this condition object, which you can then re-use within the main query